graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/ArrayPrototypeBuiltins.java

```diff
@@ -2707,6 +2713,26 @@ protected Object findIndex(Object thisObj, Object callback, Object thisArg) {
         }
     }
 
+    public abstract static class JSArrayToReversedNode extends JSArrayOperation {
+        public JSArrayToReversedNode(JSContext context, JSBuiltin builtin, boolean isTypedArrayImplementation) {
+            super(context, builtin, isTypedArrayImplementation);
+        }
+
+        @Specialization
+        protected Object reverseArray(final JSDynamicObject thisObj) {
```

Comment:
the argument type here should be `Object`. The first operation is anyway a conversion `toObject...` that makes sure that it is an object-type that JavaScript can work with. I have not checked the spec, but I assume that something like this should also work: `Array.prototype.toReversed.call("a string")`. This will magically work if you change the object type here (and would throw an `UnsupportedSpecializationException` if you don't, as there currently is not @Specialization that can handle that Type).